repair/repair.cc

```diff
@@ -2052,6 +2052,77 @@ future<> repair_service::replace_with_repair(locator::token_metadata_ptr tmptr,
     co_return co_await do_rebuild_replace_with_repair(std::move(cloned_tmptr), std::move(op), myloc.dc, reason, std::move(ignore_nodes));
 }
 
```

Comment:
> Given a table, this function repairs all the tablets belong to this
node.  If the primary_replica_only flag is set, it only repairs the
tablets if the node is the primary replica, i.e, the first node in the
locator::tablet_info.replicas.

Shouldn't it be "first node among all the nodes in tablet_info.replicas that are in this datacenter"?

Reply:
By default, repair would be global repair not a DC only repair.  This patch does not support the DC selection. If we want to support DC selection, we can extend to first node in locator::tablet_info.replicas that belongs to DC X.